demos/projects/ESPRESSIF/esp32/components/sample-azure-iot/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+set(ROOT_PATH
+    ${CMAKE_CURRENT_LIST_DIR}/../../../../../..
+)
+
+if (DEFINED CONFIG_AZURE_SAMPLE_USE_PLUG_AND_PLAY)
```

Comment:
>CONFIG_AZURE_SAMPLE_USE_PLUG_AND_PLAY [](http://example.com/codeflow?start=12&length=37)

should we have different component for each sample? 

sample-azure-iot
sample-azure-iot-pnp
sample-azure-iot-gsg

Reply:
Not sure what `sample-azure-iot-gsg` is, but the other two are mutually exclusive in this sample. We could split that into two separate samples, if that's what you are proposing.

Reply:
we have a new addition being the "getting started guide sample"

Reply:
can maybe be another PR but we should figure out the integration strategy for that sample. What defines do people select for which sample etc

Reply:
Yes, that's what we talked about earlier in the scrum. We will get this one in first, then decide about the gsg option.